modules/swagger-codegen/src/main/resources/rust/api.mustache

```diff
@@ -39,26 +40,84 @@ impl<C: hyper::client::Connect>{{classname}} for {{classname}}Client<C> {
     fn {{{operationId}}}(&self, {{#allParams}}{{paramName}}: {{#isString}}&str{{/isString}}{{#isUuid}}&str{{/isUuid}}{{^isString}}{{^isUuid}}{{^isPrimitiveType}}{{^isContainer}}::models::{{/isContainer}}{{/isPrimitiveType}}{{{dataType}}}{{/isUuid}}{{/isString}}{{#hasMore}}, {{/hasMore}}{{/allParams}}) -> Box<Future<Item = {{^returnType}}(){{/returnType}}{{#returnType}}{{{.}}}{{/returnType}}, Error = Error<serde_json::Value>>> {
         let configuration: &configuration::Configuration<C> = self.configuration.borrow();
 
+{{#hasAuthMethods}}
+        let mut auth_headers = HashMap::<String, String>::new();
+        let mut auth_query = HashMap::<String, String>::new();
+{{#authMethods}}
+{{#isApiKey}}
+        match configuration.api_key {
+            None => (),
+            Some(ref apikey) => {
+                let key = apikey.key.clone();
+                let val = match apikey.prefix {
+                    Some(ref prefix) => format!("{} {}", prefix, key),
+                    None => key,
+                };
+            {{#isKeyInHeader}}
+                auth_headers.insert("{{keyParamName}}".to_owned(), val);
+            {{/isKeyInHeader}}
+            {{#isKeyInQuery}}
+                auth_query.insert("{{keyParamName}}".to_owned(), val);
+            {{/isKeyInQuery}}
+            }
+        };
+{{/isApiKey}}
+{{#isBasic}}
+        {
+            match configuration.basic_auth {
+                Some(ref auth_conf) => {
+                    let auth = hyper::header::Authorization(
+                        hyper::header::Basic {
+                            username: auth_conf.0.to_owned(),
+                            password: auth_conf.1.to_owned(),
+                        }
+                    );
+                    auth_headers.insert("Authorization".to_owned(), format!("{}", auth));
```

Comment:
`Authorization` doesn't have a `to_string` method, just a `fmt::Display` impl. I couldn't find a cleaner way to do this.

Reply:
> This trait is automatically implemented for any type which implements the `Display` trait. As such, `ToString` shouldn't be implemented directly: `Display` should be implemented instead, and you get the `ToString` implementation for free.

https://doc.rust-lang.org/stable/std/string/trait.ToString.html

There might be some black magic involved, but as far as I know, it should have `.to_string()` since it implements the Display trait.

Reply:
Worked just fine, thanks for the suggestion/pointer!